plugins/generate.go

```diff
@@ -57,37 +40,52 @@ func main() {
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Code generated by "make plugins". DO NOT EDIT.
-
-package plugins
+`
 
-`)
+func main() {
+	data, err := os.ReadFile(filepath.Join("..", "plugins.yml"))
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	if len(plugins) == 0 {
-		return
-	}
-
-	_, err = f.WriteString("import (\n")
+	var plugins []string
+	err = yaml.Unmarshal(data, &plugins)
 	if err != nil {
 		log.Fatal(err)
 	}
 
 	for _, plugin := range plugins {
-		_, err = f.WriteString(fmt.Sprintf("\t// Register %s plugin.\n", path.Base(plugin)))
-		if err != nil {
-			log.Fatal(err)
-		}
-		_, err = f.WriteString(fmt.Sprintf("\t_ \"%s\"\n", plugin))
-		if err != nil {
+		pluginName := path.Base(plugin)
+		if err := generatePluginFile(pluginName, plugin); err != nil {
 			log.Fatal(err)
 		}
 	}
 
-	_, err = f.WriteString(")\n")
+	fmt.Printf("Generated %d plugin files\n", len(plugins))
+}
+
+func generatePluginFile(pluginName, importPath string) error {
+	filename := fmt.Sprintf("plugin_%s.go", pluginName)
+	f, err := os.Create(filename)
 	if err != nil {
-		log.Fatal(err)
+		return fmt.Errorf("failed to create %s: %w", filename, err)
+	}
+	defer f.Close()
+
+	content := fmt.Sprintf(`%s// Code generated by "make plugins". DO NOT EDIT.
+
+//go:build !remove_%s_sd
```

Comment:
Also wouldn't it be better to have option to turn  all SD off and have individual ones to turn things on? As in
```
//go:build !remove_all_sd || enable_%s_sd
```